core/shared/src/main/scala/zio/Duration.scala

```diff
@@ -179,6 +179,19 @@ final class DurationOps(private val duration: Duration) extends AnyVal {
     case _                 => ScalaFiniteDuration(duration.toNanos, TimeUnit.NANOSECONDS)
   }
 
+  def asFiniteDuration: Option[ScalaFiniteDuration] = duration match {
+    case Duration.Infinity => None
+    case Duration.Zero     => Some(ScalaDuration.Zero)
+    case _                 => Some(ScalaFiniteDuration(duration.toNanos, TimeUnit.NANOSECONDS))
+  }
+
+  def asFiniteDurationUnsafe: ScalaFiniteDuration = duration match {
```

Comment:
I don't think it's necessary to throw, because if you look at "Infinity", it's just a "really big" (biggest possible) finite duration. So I think this can and should be a total function.

Reply:
Btw I would also consider simply changing `def asScala: ScalaDuration` to `def asScala: ScalaFiniteDuration`. This way no new methods are necessary and it should be backward compatible.

Reply:
Will do